libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -616,29 +615,41 @@ internal object MapboxRouteLineUtils {
         return ranges
     }
 
-    private fun getIntersectionsWithGeometryIndex(steps: List<LegStep>?): List<StepIntersection> {
-        return steps
+    private fun getRoadClassArray(
+        steps: List<LegStep>?
+    ): Array<String?> {
+        val intersectionsWithGeometryIndex = steps
             ?.mapNotNull { it.intersections() }
             ?.flatten()
             ?.filter {
                 it.geometryIndex() != null
             }?.toList() ?: listOf()
-    }
 
-    private fun getRoadClassArray(
-        intersectionsWithGeometryIndex: List<StepIntersection>
-    ): Array<String?> {
-        return if (intersectionsWithGeometryIndex.isNotEmpty()) {
-            arrayOfNulls<String>(
-                intersectionsWithGeometryIndex.last().geometryIndex()!! + 1
-            ).apply {
-                intersectionsWithGeometryIndex.forEach {
-                    this[it.geometryIndex()!!] = it.mapboxStreetsV8()?.roadClass()
-                        ?: "intersection_without_class_fallback"
+        return when (intersectionsWithGeometryIndex.isNotEmpty()) {
+            true -> {
+                arrayOfNulls<String>(
+                    intersectionsWithGeometryIndex.last().geometryIndex()!! + 1
+                ).apply {
+                    intersectionsWithGeometryIndex.forEach {
+                        val roadClass: String =
+                            it.mapboxStreetsV8()?.roadClass()
+                                ?: "intersection_without_class_fallback"
+                        if (it.geometryIndex()!! < this.size) {
+                            this[it.geometryIndex()!!] = roadClass
+                        } else {
+                            Log.w(
+                                TAG,
+                                "Geometry index for step intersection unexpected or" +
+                                    " incorrect. Intersections are expected to have geometry " +
+                                    "indexes in ascending order."
```

Comment:
```suggestion
                                    "indexes in ascending order. There's a risk of incorrect congestion or restrictions being shown on the route."
```

Let's be super clear about what could be the implications.

Reply:
done